src/middlewared/middlewared/common/attachment/__init__.py

```diff
@@ -51,3 +57,106 @@ async def toggle(self, attachments, enabled):
         :return:
         """
         raise NotImplementedError
+
+
+class LockableFSAttachmentDelegate(FSAttachmentDelegate):
+    """
+    Represents a share/task/resource which is affected if the dataset underlying is locked
+    """
+
+    # service object
+    service_class = NotImplementedError
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.enabled_field = self.service_class.enabled_field
+        self.locked_field = self.service_class.locked_field
+        self.path_field = self.service_class.path_field
+        self.datastore_model = self.service_class._config.datastore
+        self.datastore_prefix = self.service_class._config.datastore_prefix
+        self.namespace = self.service_class._config.namespace
+        if not self.service:
+            self.service = self.service_class._config.service
+
+    async def get_query_filters(self, enabled, options=None):
+        options = options or {}
+        filters = [self.enabled_field, '=', enabled]
```

Comment:
This reads wrong, its a filter, not filters

`filters = [[self.enabled_field, '=', enabled]]`